File: android/versioned-abis/expoview-abi31_0_0/src/main/java/abi31_0_0/expo/modules/constants/ConstantsModule.java
// Copyright 2015-present 650 Industries. All rights reserved.

package abi31_0_0.expo.modules.constants;

import android.content.Context;

import java.util.Map;

import abi31_0_0.expo.core.ExportedModule;
import abi31_0_0.expo.core.ModuleRegistry;
import abi31_0_0.expo.core.Promise;
import abi31_0_0.expo.core.interfaces.ExpoMethod;
import abi31_0_0.expo.core.interfaces.ModuleRegistryConsumer;
import abi31_0_0.expo.interfaces.constants.ConstantsInterface;

public class ConstantsModule extends ExportedModule implements ModuleRegistryConsumer {
  private ModuleRegistry mModuleRegistry;

  public ConstantsModule(Context context) {
    super(context);
  }

  @Override
  public Map<String, Object> getConstants() {
    ConstantsInterface constantsService = mModuleRegistry.getModule(ConstantsInterface.class);
    return constantsService.getConstants();
  }

  @Override
  public String getName() {
    return "ExponentConstants";
  }

  @Override
  public void setModuleRegistry(ModuleRegistry moduleRegistry) {
    mModuleRegistry = moduleRegistry;
  }

  @ExpoMethod
  public void getWebViewUserAgentAsync(Promise promise) {
    String userAgent = System.getProperty("http.agent");
    promise.resolve(userAgent);
  }
}

File: android/versioned-abis/expoview-abi49_0_0/src/main/java/abi49_0_0/expo/modules/constants/ConstantsModule.kt
// Copyright 2015-present 650 Industries. All rights reserved.
package abi49_0_0.expo.modules.constants

import abi49_0_0.expo.modules.kotlin.modules.Module
import abi49_0_0.expo.modules.kotlin.modules.ModuleDefinition

class ConstantsModule : Module() {

  override fun definition() = ModuleDefinition {
    Name("ExponentConstants")

    Constants {
      return@Constants appContext.constants?.constants ?: emptyMap()
    }

    AsyncFunction("getWebViewUserAgentAsync") {
      return@AsyncFunction System.getProperty("http.agent")
    }
  }
}
